### ISSUE TITLE
Getting Error "Starting to build a new database" in Local BLAST

### ISSUE THREAD

**Issue by onurdogan** [TillMacher/apscale_gui#6]
Hi Till, 

I am getting an **Ix_xI** unexpected error **Ix_xI** when I try to use "**Build a new database**" option in **Local BLAST** section.

**1.** I'm selecting my ".**..._done.fasta**" and "**read (...table_filtered.xlsx)**" files from here as requested files:

![image](https://github.com/TillMacher/TaxonTableTools/assets/11692156/54ce97c5-81c1-429d-b47a-ce0c8049b3e3)

**2.** Then selecting **Midori2** and relevant "...gb...fasta" [file ](https://www.reference-midori.info/download.php#):

![image](https://github.com/TillMacher/TaxonTableTools/assets/11692156/2b7244f9-3b06-4c8e-8798-5ed4a6df4611)

**3.** ...and getting error like this: 
![image](https://github.com/TillMacher/TaxonTableTools/assets/11692156/3db11c96-428c-4d0e-baca-cd11c9ca010c)

4. and my "log.txt" says

> Your database: DB_MIDORI2_LONGEST_NUC_GB255_CO1_BLAST
> Your task: blastn

Is there something I am missing?

Thanks for your time and efforts in developing compatible programs.

Onur

**Comment by TillMacher** [comment]
Can retry to build the DB and press the EXIT button and tell me what is printed to the console?
For now everything looks correct from the screenshots you posted.

**Comment by onurdogan** [comment]
Hey again, here they are:

> 12:13:06: Starting to build a new database.
> Traceback (most recent call last):
>   File "/Library/Frameworks/Python.framework/Versions/3.9/bin/apscale_gui", line 33, in <module>
>     sys.exit(load_entry_point('apscale-gui==1.1.6', 'console_scripts', 'apscale_gui')())
>   File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/apscale_gui/__main__.py", line 768, in main
>     create_database_midori2(values2['database_source_file'], project_folder, databases_path)
>   File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/apscale_gui/blast_utilities.py", line 479, in create_database_midori2
>     subprocess.call([makeblastdb_exe, '-in', str(midori2_fasta), '-dbtype', 'nucl', '-out', str(db_name)])
>   File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/subprocess.py", line 349, in call
>     with Popen(*popenargs, **kwargs) as p:
>   File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/subprocess.py", line 951, in __init__
>     self._execute_child(args, executable, preexec_fn, close_fds,
>   File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/subprocess.py", line 1821, in _execute_child
>     raise child_exception_type(errno_num, err_msg, err_filename)
> FileNotFoundError: [Errno 2] No such file or directory: 'makeblastdb

**Comment by TillMacher** [comment]
It seems like you do not have the blast+ software installed. APSCALE is looking for the 'makeblastdb' command but cannot find it.  Could you please install [blast+](https://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/) and then try it again?

cheers
Till

**Comment by onurdogan** [comment]
Heya,

Thanks for pointed out this. Actually, it had already been installed before, but it was holding my download directory instead of intended location "/Usr/local/bin". This is likely why it didn't work . After moving it to the "bin" folder, I encountered a security alert window from the apscale_gui because it thought that "makeblastdb" had been installed from an untrustworthy source. To resolve this, I provided some permissions, and it worked properly afterwards.

In summary, I did something, and it worked well thanks.

Onur

**Comment by onurdogan** [comment]
Hey again,

I encountered an error while trying to create a **new** database in the same location where I had previously created a successful one -this older one is still functioning efficiently.

**Steps:**
**Load Requires Files**
1. I've provided both "...fasta" and "...OTU_filtered.xlsx (as Read file)" files as required input files, which are located in the "9_lulu_filtering/otu_clustering" folder.

**Build a new database**
1. I've selected "Midori2" as a format.
2. I've selected a new database source file named [MIDORI2_UNIQ_NUC_GB257_CO1_BLAST.fasta] which is locating in the APSCALE_Databases folder.
3. Build a new database [Go]

then...

onurdogan@Onurs-iMac ~ % apscale_gui
Traceback (most recent call last):
  ```
File "/Library/Frameworks/Python.framework/Versions/3.9/bin/apscale_gui", line 8, in <module>
    sys.exit(main())
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/apscale_gui/__main__.py", line 772, in main
    create_database_from_fasta(values2['database_source_file'], project_folder, databases_path)
TypeError: create_database_from_fasta() missing 1 required positional argument: 'databases_path'
```
Enjoy,
